File: src/ProjectR/content/scripts/spells/SniffMyFinger.lua
function GetTargetType()
    return Single
end

function GetName()
    return "Sniff my finger!"
end

function GetDescription()
    return "You know exactly what this does. Reduces target DEF + MR and restores some MP."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return 0
end

function GetDelay()
  return .5
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
    d:BuffStat(DEF, -.2)
    d:BuffStat(MR, -.2)
    a:UseMP(-(aMP * .2))
end

File: src/ProjectR.Scripting/content/scripts/spells/SniffMyFinger.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class SniffMyFinger : SpellScriptBase
    {
        public override string Name { get { return "Sniff my finger!"; } }
        public override string Description { get { return "You know exactly what this does. Reduces target DEF + MR and restores some MP."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 0; } }
        public override double Delay { get { return .5; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            BuffStat(Stat.DEF, -.3);
            BuffStat(Stat.MR, -.3);
            caster.UseMP(-40);
        }
    }
}